CHANGELOG.md

```diff
@@ -1,10 +1,43 @@
+* The entire BSON package used to be re-exported as Realm.BSON, to simplify the new SDK we want to export only the BSON types that our SDK database component supports (ObjectId, Decimal128 and UUID). See #4934.
+* We're now reusing code to perform assertions and although this is strictly not a breaking change, since we havn't historically documented error messages, you should probably revisit any code in your app which relies on matching on specific error messages.
+* `Results`, `List` and `Set` used to inherit directly from `Collection` but now inherits from an abstract `OrderedCollection`, which interns extends `Collection`.
+* Users will no longer be able to mix declaring schemas using a shorthand (string) (ex “object[]” for a list of objects) and declaring an “objectType” in their schema. They have to choose one and stick with it for every property. This is a minor breaking change since this used to be allowed:
+```js
+{ type: "object[]", objectType: "SomeObject" } // No longer works
+{ type: "list", objectType: "SomeObject" } // This still works
+```
+* To prevent modifying end-users class-based model classes, we’re now wrapping class-based models in our own model. This is technically not a breaking change, as the previous behavior was undocumented and while objects will still pass `instanceof SomeClass` checks, code who is testing using prototype comparisons will fail:
+```js
+Object.getPrototypeOf(object) == CustomObject.prototype // No longer works
+```
+* It used to be the case that Symbols got type coerced to string when used as keys into a dictionary, this is undocumented behaviour that we were testing for which makes little sense in practice, hence won’t be implemented in the new SDK.
+* As a part of migrating to NAPI (since ~ v6), we saw no performant way to support getting property names of a Realm.Object via the standard Object.keys(obj). As a side-effect we stopped supporting the object spread operator `{...obj}` and `Realm.Object#keys()`, `Realm.Object#entries()` and `Realm.Object#toJSON()` methods were introduced as a workaround. The new SDK wraps its accessor objects in a Proxy trapping the ownKeys trap which enables calls to the standard `Object.keys(obj)` and the spread operator `{...obj}` and therefore we suggest deprecating the API with the @deprecation (plus console.warn once) of RealmObject#keys() and RealmObject#entries(). RealmObject#toJSON still serves the purpose of producing a circularly referencing object graph. We would love the community's feedback on this!
```

Comment:
```suggestion
* As a part of migrating to NAPI (since ~ v6), we saw no performant way to support getting property names of a Realm.Object via the standard Object.keys(obj). As a side-effect we stopped supporting the object spread operator `{...obj}` and `Realm.Object#keys()`, `Realm.Object#entries()` and `Realm.Object#toJSON()` methods were introduced as a workaround. The new SDK wraps its accessor objects in a Proxy trapping the ownKeys trap which enables calls to the standard `Object.keys(obj)` and the spread operator `{...obj}`. Therefore, we are deprecating the API with the @deprecation flag and a single console.warn of RealmObject#keys() and RealmObject#entries(). RealmObject#toJSON still serves the purpose of producing a circularly referencing object graph. We would love the community's feedback on this!
```

Reply:
I've created https://github.com/realm/realm-js/issues/5605 to track the actual deprecation as I'm actually unsure if we ever did that.